src/assignmentNotification.js

```diff
@@ -1,49 +1,60 @@
-            throw new Error('must provide assignment');
-        }
-    };
-
-    _AssignmentNotification.prototype.send = function() {
-        // FIXME: The current implementation of this requires 2 HTTP requests
-        // to guarantee that the server is notified of the assignment. By decoupling
-        // the assignment notification from the analytics write success we can
-        // bring this down to 1 HTTP request
-
-        this.persistAssignment();
-
-        this._visitor.analytics.trackAssignment(
-            this._visitor.getId(),
-            this._assignment,
-            function(success) {
-                this.persistAssignment(success ? 'success' : 'failure');
-            }.bind(this));
-    };
-
-    _AssignmentNotification.prototype.persistAssignment = function(trackResult) {
-        return $.ajax(TestTrackConfig.getUrl() + '/api/v1/assignment_event', {
-            method: 'POST',
-            dataType: 'json',
-            crossDomain: true,
-            data: {
-                visitor_id: this._visitor.getId(),
-                split_name: this._assignment.getSplitName(),
-                context: this._assignment.getContext(),
-                mixpanel_result: trackResult
-            }
-        }).fail(function(jqXHR, textStatus, errorThrown) {
-            var status = jqXHR && jqXHR.status,
-                responseText = jqXHR && jqXHR.responseText;
-            this._visitor.logError('test_track persistAssignment error: ' + [jqXHR, status, responseText, textStatus, errorThrown].join(', '));
+import $ from 'jquery';
+import TestTrackConfig from './testTrackConfig';
+
+var AssignmentNotification = function(options) {
+    options = options || {};
+    this._visitor = options.visitor;
+    this._assignment = options.assignment;
+
+    if (!this._visitor) {
+        throw new Error('must provide visitor');
+    } else if (!this._assignment) {
+        throw new Error('must provide assignment');
+    }
+};
+
+AssignmentNotification.prototype.send = function() {
+    // FIXME: The current implementation of this requires 2 HTTP requests
+    // to guarantee that the server is notified of the assignment. By decoupling
+    // the assignment notification from the analytics write success we can
+    // bring this down to 1 HTTP request
+
+    var firstPersist = this._persistAssignment()
+
+    var secondPersist = $.Deferred();
+    this._visitor.analytics.trackAssignment(
+        this._visitor.getId(),
+        this._assignment,
+        function(success) {
+            var promise = this._persistAssignment(success ? 'success' : 'failure');
+            promise.always(function() {
+                if (promise.state() === 'resolved') {
+                    secondPersist.resolve();
+                } else {
+                    secondPersist.reject();
+                }
+            });
         }.bind(this));
-    };
 
-    return _AssignmentNotification;
-})();
+    return $.when(firstPersist, secondPersist);
```

Comment:
this change is for testing. i want to know when the persist calls completes, so now it returns a promise.